### ISSUE TITLE
Failed assertion in i965 output with 0.11

### ISSUE THREAD

**Issue by yunake** [mpv-player/mpv#2388]
Today I encountered the following crash:

```
yunake@x230:~$ mpv https://www.youtube.com/watch?v=kb146Y1igTQ
Playing: https://www.youtube.com/watch?v=kb146Y1igTQ
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=eng (*) (aac)
libva info: VA-API version 0.38.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_38
libva info: va_openDriver() returns 0
xscreensaver-command: not active: idle timer reset.

AO: [pulse] 44100Hz stereo 2ch float
Using hardware decoding (vaapi).
VO: [vaapi] 1280x720 vaapi

<skip>
(Paused) AV: 00:08:36 / 00:10:06 (85%) A-V: -0.001 DS: +0.13404% Missed: 155 Dropped: 110 Cache:  9s+13MB
[input] No key binding found for key '0x21ff03'.
mpv: i965_output_dri.c:167: i965_put_surface_dri: Assertion `dest_region->bo' failed.e:  9s+13MB
```

JFYI, I use `video-sync=display-resample`, `vo=vaapi` & `hwdec=vaapi`. This is a fully-updated ArchLinux system.

I tried to reproduce under gdb to get a core/bt, but was unable to with the limited tests I did so far. How can I help you to debug this further?

**Comment by wm4** [comment]
It's rather unlikely that this is mpv's fault. A driver bug is much more likely.

**Comment by yunake** [comment]
you are probably right. i managed to get a core file via gdb just now, here's just the backtrace:
```
#0  0x00007fffefd845f8 in raise () from /usr/lib/libc.so.6
#1  0x00007fffefd85a7a in abort () from /usr/lib/libc.so.6
#2  0x00007fffefd7d417 in __assert_fail_base () from /usr/lib/libc.so.6
#3  0x00007fffefd7d4c2 in __assert_fail () from /usr/lib/libc.so.6
#4  0x00007fffd4191081 in ?? () from /usr/lib/dri/i965_drv_video.so
#5  0x00007fffd416626c in ?? () from /usr/lib/dri/i965_drv_video.so
#6  0x0000555555657bed in ?? ()
#7  0x000055555565256c in ?? ()
#8  0x00007ffff2e224a4 in start_thread () from /usr/lib/libpthread.so.0
#9  0x00007fffefe3a13d in clone () from /usr/lib/libc.so.6
```
I can trigger it consistently by just going fullscreen <-> windowed multiple times. I will report this to the Intel driver devs.

**Comment by cranes-bill** [comment]
I can't play this directly from the stream, because **MPV** says:
 _„[ffmpeg] tls: The TLS connection was non-properly terminated.
Failed to recognize file format.“_ , 
but if i download the video and then play it, everything looks fine (i think):
```
mpv 'Metabolism _ Nutrition, part 2 Crash Course A_P 37.mp4' 
Playing: Metabolism _ Nutrition, part 2 Crash Course A_P 37.mp4
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=eng (*) (aac)
libva info: VA-API version 0.38.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_38
libva info: va_openDriver() returns 0
AO: [pulse] 44100Hz stereo 2ch float
Using hardware decoding (vaapi).
VO: [vaapi] 1280x720 vaapi
AV: 00:00:11 / 00:10:06 (1%) A-V:  0.000


Exiting... (Quit)

```

**Comment by yunake** [comment]
this was not a problem with the media (no idea why you thought it was). the problem was fixed in libva-intel 1.6.1
